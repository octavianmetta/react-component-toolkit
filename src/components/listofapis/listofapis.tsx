// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

import React from "react";
import styled from "styled-components";

import { listofapisProps } from "./listofapis.types";

const Styledlistofapis = styled.div<listofapisProps>`
  color: #000;
  text-align: center;
`;

const listofapis: React.FC<listofapisProps> = ({ ...props }) => {
  return (
    <>
      <Styledlistofapis {...props} data-testid="listofapisTestId">
        {props.message}
      </Styledlistofapis>
    </>
  );
};

export default listofapis;
