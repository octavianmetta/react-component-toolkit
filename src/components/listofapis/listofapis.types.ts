// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

export interface listofapisProps {
    id: string,
    message: string
}