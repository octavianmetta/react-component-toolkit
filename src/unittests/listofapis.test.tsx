// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import listofapis from "../components/listofapis/listofapis";

describe("Testing component : <listofapis />", () => {
  const componentId = "ComponentUnderTest";

  beforeAll(() => {
  });

  afterAll(() => {
  })
  
  beforeEach(() => {
  });

  afterEach(() => {
  });

  test("<listofapis /> should render with id", async () => {

    // Create our component with our generated id
    const testRenderer = renderer.create(<listofapis message="" id={componentId} />);

    // Use the created renderer to convert to json and then check it matches our expected snapshot
    const jsonSnapshot = testRenderer.toJSON();

    // Check that our rendered component does have the exact randomly generated componentid for this test
    expect(jsonSnapshot).toHaveProperty("props.id", componentId);

    // Match that our rendered component matches our snapshot of the component and ignore the random id
    expect(jsonSnapshot).toMatchSnapshot(
      {
        props: { id: expect.any(String) }
      }
    );
  });
});
