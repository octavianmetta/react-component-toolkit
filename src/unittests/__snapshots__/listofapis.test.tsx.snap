// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component : <listofapis /> <listofapis /> should render with id 1`] = `
{
  "children": null,
  "props": {
    "className": "sc-bcXHqe ileYXv",
    "data-testid": "listofapisTestId",
    "id": Any<String>,
  },
  "type": "h1",
}
`;
